h/browser/chrome/lib/uri-info.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * A wrapper for the server's uriinfo endpoint.
+ *
+ * Exports a function that makes an HTTP request to the server's uriinfo
+ * endpoint and returns a Promise that resolves to the response. Usage:
+ *
+ *     var uriInfo = require('uri-info');
+ *     uriInfo(uri).then(function(info) {
+ *       ...
+ *     });
+ *
+ * This function can be called with the same uri multiple times consecutively
+ * and it'll just keep returning the same Promise - won't make multiple HTTP
+ * requests.
+ *
+ * But if you call the function with a different uri, then call it with the
```

Comment:
Because the promise is a singleton shared by all tabs, if I open multiple tabs to the same URL consecutively it'll only send one request. If I open multiple tabs to different URLs it'll send multiple requests.

I think I might have to investigate what happens when we send lots of simultaneous requests for different tabs with different URLs, that should be showing different badge numbers or be blocked/not blocked...